### PR TITLE
Docs Search

### PR DESCRIPTION
**Pull request by coffee-cup**
Add search to docs that will filter the sidebar based on your query. Right now it only searches based on the doc title and slug, but we can easily extend this if we want.

docs-search

### CODE REVIEW

**Comment by andrejak** [comment]
Are we planning to add full text search at some point? For example, if I want to find "createUniverseWatcher" documentation, this isn't really helping me right now. (Otherwise looks good to me)

File: docs/src/components/Sidebar.tsx [hunk 1]
```diff
@@ -147,19 +165,32 @@ export interface Props {
...
         <Section
           key={section.slug}
           section={section}
+          searching={isSearching}
           active={currentPath.startsWith(section.slug)}
           currentPath={currentPath}
         />
       ))}
+
+      {filter != null && filter.length === 0 && <p>{"No Results :("}</p>}
```
Comment (andrejak) [inline docs/src/components/Sidebar.tsx]: I'm slightly confused by the variable name filter here. Does it mean filtered results, or terms to filter out? 
Should this line be a check if results/sections are empty?

Comment (coffee-cup) [reply docs/src/components/Sidebar.tsx]: `filter` is either `null` or a list of document ids that should be displayed. It is basically the results. I'll rename the variable.

**Comment by coffee-cup** [comment]
@andrejak No idea why those results are there. I'm just using https://www.gatsbyjs.org/packages/gatsby-plugin-local-search/ which uses an algorithm called [flexsearch](https://github.com/nextapps-de/flexsearch).

**Comment by coffee-cup** [comment]
@andrejak full text search is possible, but then there are way too many results. For example, search plugin will bring up most of the results (because that word is mentioned in a lot of the docs).

I've changed it so that headings in docs are also searched. I think it is a bit better, but could also be too verbose. Can you please lmk what you think.

**Pull request merged**